### ISSUE TITLE
(RFU) Pokemon Emerald

### ISSUE THREAD

**Issue by AeolusUX** [libretro/gpsp#247]
Union room works flawlessly! I'm just wondering if you guys tested the direct corner colosseum and trade center? I've tried both and seems it's disconnecting and giving me a communication error. Would this be the said limitation from david's blog, because the game doesn't like high latency? or the emulation of packet loss? I am playing using lan.

**Comment by ysheng26** [comment]
I haven't tried the Union room but I tried to trade on RG35XX Plus 10.1 and it had this communication error as well. Best I got was to get to the trading room but when I sit down it disconnected

**Comment by papajefe** [comment]
Discovered this recently and did some testing with a friend today.

We also experienced seemingly random disconnects from the Client player. Each of us tried hosting and using the "Relay Server" features in RetroArch 1.17

Tested Union Room, Trade Center and Colosseum modes.

Union Room itself was fine, no "Connection Error" in general use.
Trade Center was able to complete a Trade once, but then "Connection Error" after.
Colosseum would usually "Connection Error" during battle load, but was able to get a few turns in sometimes.

Overall VERY cool feature that I hope to see improve as time goes on. From reading some of DavidGFs writings on the subject it sounds like at least for Pokemon Fire Red\Leaf Green\Emerald we can achieve 100% reliable netplay. Not sure about games with more real-time requirements.

**Comment by cvharris** [comment]
same thing found today. Colosseum crashes almost immediately, if not soon into the battle. Same Retroarch and gpSP versions on both devices: 1.19.1 & v1.0.0-f2837be respectively

**Comment by HaydenBYU** [comment]
I am having a similar issue but with FireRed. I keep getting communication errors at random times. I was able to see my friends trainer badge but then it crashed as we started a battle.
<img width="722" alt="Screenshot 2024-10-25 at 12 36 34 PM" src="https://github.com/user-attachments/assets/52ee036d-50d3-466e-bb07-47466bdf4d34">

**Comment by Bkrouse95** [comment]
I have experienced the same communication errors as everyone above. Very cool though.

**Comment by xanat0s** [comment]
Yeah this issue has a really bad title, it has nothing to do with Emerald.

I've heard the gameboy link is hyper sensitive to desync but I'm not a hardware guy so I don't know how accurate that is. If true, that's the most likely cause and is something the devs here can do little about. It would also mean it'd work far more reliably on a local network than, say, with someone across the country. 

Experiencing this on LeafGreen in the Union Room, Colosseum, and Trade Center.

**Comment by dejalavidavolar** [comment]
same error here, tested on emerald and hackroms

**Comment by Bkrouse95** [comment]
Been testing, have found that best results are when you can connect 2 devices on the same LAN and have a ping under 20.

**Comment by AeolusUX** [comment]
Thanks for the insight! We’re still using gpsp on LAN and adjusted the check frame setting to 10, which reduced desyncs a lot for Direct Corner. It’s somewhat stable now but still disconnects occasionally on the client side. I haven't tried again recently I'm guessing there have been improvements since I originally posted this issue.
